### PR TITLE
Prmdr 163 migrate authorisation handler

### PR DESCRIPTION
**Pull request by joefong-nhs**
- Add an authoriser lambda which verify and decode a JWT authorizationToken, and respond with IAM policy which allow or deny user access to other APIs
- Add unit test for above lambda
- Change `make zip` command to pack with `--platform manylinux2014_x86_64` flag, as python's cryptography package seems to require some platform specific native library

### CODE REVIEW

File: Makefile
Comment (SRAlexander) [inline Makefile:49]: We should give this a good test, but it sounds like a reasonable adjustment.

File: lambdas/handlers/authoriser_handler.py
Comment (SRAlexander) [inline lambdas/handlers/authoriser_handler.py:53]: can we rename this to something more meaningful?

File: lambdas/handlers/authoriser_handler.py
Comment (SRAlexander) [inline lambdas/handlers/authoriser_handler.py:27]: How are we planning to set the os.env on the aws lambda instance?

**Review by SRAlexander — CHANGES_REQUESTED** [review]
Looks good, we will have to look at the deployment of lambda but we will have to wait until we can merge the infrastructure side.

File: lambdas/handlers/authoriser_handler.py
Comment (joefong-nhs) [reply lambdas/handlers/authoriser_handler.py]: Thanks!  Addressed in this commit: https://github.com/nhsconnect/national-document-repository/pull/42/commits/c0192931217fdc6f88a751322e92b745595cfa75

File: lambdas/handlers/authoriser_handler.py
Comment (joefong-nhs) [reply lambdas/handlers/authoriser_handler.py:27]: In a branch for subtask 165, we added this env var to the lambda by terraform : https://github.com/nhsconnect/national-document-repository-infrastructure/blob/prmdr_165_subtask/infrastructure/lambda-authoriser.tf

**Pull request merged**
